bindings_node/src/mls_client.rs

```diff
@@ -0,0 +1,209 @@
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(RecoverableEcdsaSignature::new(
+      signature_text,
+      signature_bytes.deref().to_vec(),
+    ));
+
+    signature_request
+      .add_signature(signature)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    self
+      .inner_client
+      .register_identity(signature_request)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    return Ok(());
+  }
+
+  #[napi]
+  pub async fn register_erc1271_identity(
+    &self,
+    signature_bytes: Uint8Array,
+    chain_rpc_url: String,
+  ) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(
+      Erc1271Signature::new_with_rpc(
+        signature_text,
+        signature_bytes.deref().to_vec(),
+        self.account_address.clone(),
+        chain_rpc_url,
+      )
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?,
+    );
+
+    signature_request
+      .add_signature(signature)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    self
+      .inner_client
+      .register_identity(signature_request)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    return Ok(());
+  }
+
+  #[napi]
+  pub fn signature_text(&self) -> Option<String> {
+    match self.inner_client.identity().signature_request() {
+      Some(signature_req) => Some(signature_req.signature_text()),
+      None => None,
+    }
+  }
+
+  #[napi]
+  pub fn release_db_connection(&self) -> Result<()> {
+    Ok(
+      self
+        .inner_client
+        .release_db_connection()
+        .map_err(|e| Error::from_reason(format!("{}", e)))?,
+    )
+  }
+
+  #[napi]
+  pub async fn db_reconnect(&self) -> Result<()> {
+    Ok(
+      self
+        .inner_client
+        .reconnect_db()
+        .map_err(|e| Error::from_reason(format!("{}", e)))?,
+    )
+  }
```

Comment:
i saw these DB functions in the FFI bindings. not sure if they're necessary for Node.

Reply:
I used `git blame` to find the PR that added this method in the FFI bindings: https://github.com/xmtp/libxmtp/pull/759

It seems that this won't be relevant for Node, seeing as you don't have the app backgrounding issue that iOS has